pkg/dataconn/client.go

```diff
@@ -153,14 +114,77 @@ func (c *Client) Close() {
 
 func (c *Client) loop() {
 	defer close(c.send)
+	ticker := time.NewTicker(1 * time.Second)
```

Comment:
Looks good to me.  I don't have much of a preference between using the Timer versus Ticker, but I disliked how ugly stopping the Timer needs to be. The Timer's API seems more oriented towards something that is used once instead of something that is constantly stopped and reset.   I don't think the loss of accuracy is important.